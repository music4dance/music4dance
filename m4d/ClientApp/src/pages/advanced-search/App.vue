<script setup lang="ts">
import { DanceQuery } from "@/models/DanceQuery";
import { SongFilter } from "@/models/SongFilter";
import { SongSort, SortOrder } from "@/models/SongSort";
import { Tag, TagContext } from "@/models/Tag";
import { UserQuery } from "@/models/UserQuery";
import { getMenuContext } from "@/helpers/GetMenuContext";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { safeTagDatabase } from "@/helpers/TagEnvironmentManager";
import { computed, ref } from "vue";
import type { DanceDatabase } from "@/models/DanceDatabase/DanceDatabase";
import { DanceQueryItem } from "@/models/DanceQueryItem";
import TagQuerySelector from "./components/TagQuerySelector.vue";

const context = getMenuContext();
const danceDB: DanceDatabase = safeDanceDatabase();

const allDances = danceDB.all;

const tagDatabase = safeTagDatabase();
const filter = getQueryFilter();
const danceQueryInit = new DanceQuery(filter.dances);

const showDiagnostics = getShowDiagnostics();
const keyWords = ref(filter.searchString ?? "");
const advancedText = ref(false);
const danceQueryItems = ref(danceQueryInit.danceQueryItems);
const danceConnector = ref(danceQueryInit.isExclusive ? "all" : "any");
const dances = computed<string[]>({
  get: () => danceQueryItems.value.map((d) => d.id),
  set: (value: string[]): void => {
    // Sync danceQueryItems with selected dances, preserving thresholds and tags
    danceQueryItems.value = value.map((id) => {
      const existing = danceQueryItems.value.find((d) => d.id === id);
      return existing ? existing : new DanceQueryItem({ id: id, threshold: 1 });
    });
  },
});
const hasDanceDetails = computed(() => {
  return danceQueryItems.value.some((d) => d.threshold > 1 || d.tagQuery?.hasTags);
});
const showDanceDetails = ref(hasDanceDetails.value);
const tags: Tag[] = tagDatabase.tags;

// Initialize tagString and excludeDanceTags from filter
let initialTagString = filter.tags ?? "";
const excludeDanceTags = ref(false);

// Parse initial state from tagString: if it starts with "^", exclude dance_ALL tags
if (initialTagString.startsWith("^")) {
  excludeDanceTags.value = true;
  initialTagString = initialTagString.substring(1);
}

const tagString = ref(initialTagString);

// Check if any style tags are currently selected
const hasStyleTags = computed(() => {
  if (!tagString.value) return false;
  const parts = tagString.value.split("|").map((p) => p.trim());
  const styleTags = tags.filter((tag) => tag.category.toLowerCase() === "style");
  const styleKeys = styleTags.map((tag) => tag.key);

  return parts.some((part) => {
    const cleanPart = part.startsWith("+") || part.startsWith("-") ? part.slice(1) : part;
    return styleKeys.includes(cleanPart);
  });
});

const tempoMin = ref(filter.tempoMin ?? 0);
const tempoMax = ref(filter.tempoMax ?? 400);

const lengthMin = ref(filter.lengthMin ?? 0);
const lengthMax = ref(filter.lengthMax ?? 600);

const userQueryInit = new UserQuery(filter.user);

const user = ref(userQueryInit.userName ?? context.userName ?? "");
const displayUser = ref(userQueryInit.displayName);

const bonuses = ref(computeBonuses());
const validated = ref(false);
const services = ref(filter.purchase ? filter.purchase.trim().split("") : []);
const activity = ref(userQueryInit.parts);

const songFilter = computed(() => {
  const danceQuery = DanceQuery.fromParts(
    danceQueryItems.value.map((t) => t.toString()),
    danceConnector.value === "all",
  );
  const userQuery = UserQuery.fromParts(
    computedActivity.value ? computedActivity.value : undefined,
    isAnonymous.value ? user.value : displayUser.value,
  );
  const filter = new SongFilter();
  let level = 0;
  if (bonuses.value.indexOf("P") !== -1) {
    level = 1;
  }
  if (bonuses.value.indexOf("D") !== -1) {
    level += 2;
  }

  filter.action = "advanced";
  filter.searchString = keyWords.value;
  filter.dances = danceQuery.query;
  filter.sortOrder = SongSort.fromParts(sortId.value ?? undefined, sortDirection.value).query;
  filter.user = userQuery.query;
  filter.purchase = services.value.join("");
  filter.tempoMin = tempoMin.value === 0 ? undefined : tempoMin.value;
  filter.tempoMax = tempoMax.value >= 400 ? undefined : tempoMax.value;
  filter.lengthMin = lengthMin.value === 0 ? undefined : lengthMin.value;
  filter.lengthMax = lengthMax.value >= 400 ? undefined : lengthMax.value;
  // Add "^" prefix if dance_ALL tags are excluded
  filter.tags = excludeDanceTags.value ? `^${tagString.value}` : tagString.value;
  filter.level = level ? level : undefined;

  return filter;
});

const sortOptions = computed(() => [
  { text: computedDefault(), value: null },
  { text: "Dance Rating", value: SortOrder.Dances },
  { text: "Closest Match", value: SortOrder.Match },
  { text: "Title", value: SortOrder.Title },
  { text: "Artist", value: SortOrder.Artist },
  { text: "Tempo", value: SortOrder.Tempo },
  { text: "Length", value: SortOrder.Length },
  { text: "Last Modified", value: SortOrder.Modified },
  { text: "Last Edited", value: SortOrder.Edited },
  { text: "When Added", value: SortOrder.Created },
  { text: "Energy", value: SortOrder.Energy },
  { text: "Mood", value: SortOrder.Mood },
  { text: "Strength of Beat", value: SortOrder.Beat },
  { text: "Comments", value: SortOrder.Comments },
]);

const sortInit = new SongSort(filter.sortOrder, filter.TextSearch);
const sortId = ref(sortInit.id || null);
const sortDirection = ref(sortInit.direction);

const danceNames = computed(() => {
  return dances.value.map((d) => danceDB.danceFromId(d)!.name);
});

const activities = computed(() => {
  const user = displayUser;
  const empty = { text: "Don't filter on user activity", value: "NT" };

  if (!user.value) {
    return [empty];
  }

  const my = user.value === "me" ? "my" : user.value + "'s";
  const i = user.value === "me" ? "I have" : user.value + " has";

  const items = [
    { text: `Include all songs in ${my} favorites`, value: "IL" },
    { text: `Exclude all songs in ${my} favorites`, value: "XL" },
    { text: `Include all songs ${i} tagged`, value: "IT" },
    { text: `Exclude all songs ${i} tagged`, value: "XT" },
    { text: `Exclude all songs in ${my} blocked list`, value: "XH" },
    { text: `Include all songs in ${my} blocked list`, value: "IH" },
  ];

  if (dances.value.length > 0) {
    items.unshift({
      text: `Include all songs ${i} voted against ${danceNames.value.join(", ")}`,
      value: "IX",
    });
    items.unshift({
      text: `Include all songs ${i} voted for ${danceNames.value.join(", ")}`,
      value: "ID",
    });
  }
  items.unshift(empty);
  return items;
});

const computedActivity = computed({
  get: function (): string {
    return displayUser.value ? activity.value : "NT";
  },
  set: function (value: string): void {
    if (displayUser.value) {
      activity.value = value;
    }
  },
});

const isAnonymous = computed(() => {
  return displayUser.value === "Anonymous";
});

function getQueryFilter(): SongFilter {
  const params = new URLSearchParams(window.location.search);
  const filterString = params.get("filter");

  return filterString ? SongFilter.buildFilter(filterString) : new SongFilter();
}

function getShowDiagnostics(): boolean {
  const params = new URLSearchParams(window.location.search);
  const showDiagnostics = params.get("showDiagnostics") ?? params.get("showdiagnostics");

  return !!showDiagnostics && showDiagnostics !== "false";
}

const computedDefault = (): string => {
  const value = !!keyWords.value ? "Closest Match" : "Dance Rating";
  return `Default (${value})`;
};

function computeBonuses(): string[] {
  const bonuses = [];
  if (filter.level && filter.level & 1) {
    bonuses.push("P");
  }
  if (filter.level && filter.level & 2) {
    bonuses.push("D");
  }
  return bonuses;
}

async function onSubmit(): Promise<void> {
  const form = document.getElementById("advanced-search") as HTMLFormElement;

  if (form.checkValidity() === true) {
    const min = tempoMin.value;
    const max = tempoMax.value;
    if (min > max) {
      tempoMax.value = min;
      tempoMin.value = max;
    }

    if (lengthMin.value > lengthMax.value) {
      const length = lengthMax.value;
      lengthMax.value = lengthMin.value;
      lengthMin.value = length;
    }

    const loc = window.location;
    const query = songFilter.value.encodedQuery;

    const state = window.location.pathname + `?filter=${query}`;
    window.history.replaceState(null, "", state);

    window.location.href = `${loc.origin}/song/filtersearch?filter=${query}`;
  }

  validated.value = true;
}

function onReset(evt: Event): void {
  evt.preventDefault();
  const userName = isAnonymous.value ? user : displayUser;

  keyWords.value = "";
  advancedText.value = false;
  dances.value = [];
  danceConnector.value = "any";
  tempoMin.value = 0;
  tempoMax.value = 400;
  lengthMin.value = 0;
  lengthMax.value = 600;
  if (userName) {
    user.value = userName.value;
    activity.value = "IH";
  } else {
    user.value = "";
    activity.value = "NT";
  }
  displayUser.value = "";
  services.value = [];
  sortId.value = null;
  sortDirection.value = "asc";
  bonuses.value = [];
  tagString.value = "";
  excludeDanceTags.value = false; // Reset dance_ALL tags toggle

  // Reset danceQueryItems to empty
  danceQueryItems.value = [];

  validated.value = false;
}
</script>

<template>
  <PageFrame id="app">
    <h1 class="col-sm-12" style="font-size: 22px; text-align: center">Advanced Song Search</h1>
    <div style="max-width: 600px; margin-left: auto; margin-right: auto">
      <BForm
        id="advanced-search"
        :validated="validated"
        novalidate
        @submit.stop.prevent="onSubmit"
        @reset="onReset"
      >
        <KeywordEditor
          id="search-string-group"
          v-model="keyWords"
          v-model:advanced="advancedText"
        />

        <BFormGroup id="dance-group" label="Dances:">
          <div style="border: 1px solid #ced4da; border-radius: 0.25rem">
            <DanceSelector id="dance-selector" v-model="dances" :dance-list="allDances" />
            <div class="d-flex mb-2">
              <BFormRadioGroup
                id="dance-connector"
                v-model="danceConnector"
                buttons
                button-variant="outline-secondary"
                :disabled="dances.length < 2"
                class="mx-3"
                size="sm"
              >
                <BFormRadio value="any">Any</BFormRadio>
                <BFormRadio value="all">All</BFormRadio>
              </BFormRadioGroup>
              <BFormCheckbox
                id="show-dance-details"
                v-model="showDanceDetails"
                :disabled="!dances.length || hasDanceDetails"
                switch
              >
                Show Dance Details
              </BFormCheckbox>
            </div>
            <div v-if="showDanceDetails" class="mx-3 mb-2">
              <div
                v-for="item in danceQueryItems"
                :key="item.id"
                class="border rounded p-3 mt-2 bg-light"
              >
                <!-- Dance name header -->
                <h6 class="mb-2 text-primary fw-bold">
                  {{ item.dance.name }}
                </h6>

                <!-- Threshold control -->
                <BFormGroup
                  :label="`Minimum ${item.dance.name} rating:`"
                  :label-for="`sb-${item.id}`"
                  class="mb-2"
                >
                  <BFormSpinbutton
                    :id="`sb-${item.id}`"
                    v-model="item.threshold"
                    inline
                    min="1"
                    max="30"
                    size="sm"
                    class="d-inline-block"
                  />
                  <small class="text-muted ms-2">
                    ({{ item.threshold === 1 ? "Any rating" : `${item.threshold}+ votes` }})
                  </small>
                </BFormGroup>

                <!-- Dance-specific tag selector -->
                <BFormGroup :label="`${item.dance.name}-specific tags:`" class="mb-0">
                  <TagQuerySelector
                    :model-value="item.tagQuery?.tagList.summary ?? ''"
                    :tag-list="tags"
                    :context="TagContext.Dance"
                    @update:model-value="
                      (val) => {
                        if (val && val.length > 0) {
                          item.tags = val;
                        } else {
                          item.tags = undefined;
                        }
                      }
                    "
                  />
                </BFormGroup>
              </div>
            </div>
          </div>
        </BFormGroup>

        <!-- Song-level Tags Section -->
        <BFormGroup label="Song Tags:">
          <div style="border: 1px solid #ced4da; border-radius: 0.25rem" class="p-3">
            <TagQuerySelector
              v-model="tagString"
              :tag-list="tags"
              :context="excludeDanceTags ? TagContext.Song : [TagContext.Song, TagContext.Dance]"
            />

            <div class="row mt-3">
              <div class="col-6">
                <BFormCheckbox
                  id="exclude-dance-tags"
                  v-model="excludeDanceTags"
                  :disabled="hasStyleTags"
                  switch
                >
                  Exclude Dance Tags
                </BFormCheckbox>
              </div>
              <div class="col-6">
                <small class="text-muted">
                  Only use song tags (do not include tags from any dance)
                </small>
              </div>
            </div>
          </div>
        </BFormGroup>

        <BFormGroup id="tempo-range-group" label="Tempo range (BPM):" label-for="tempo-range">
          <BFormGroup id="tempo-range">
            <div class="d-flex">
              <BFormInput
                id="tempo-min"
                v-model.number="tempoMin"
                type="number"
                min="0"
                max="400"
                style="width: 6rem"
              />
              <span class="mx-2 pt-2">to</span>
              <BFormInput
                id="tempo-max"
                v-model.number="tempoMax"
                type="number"
                min="0"
                max="400"
                style="width: 6rem"
              />
              <div class="invalid-feedback">Tempos must be between 0 and 400 BPM</div>
            </div>
          </BFormGroup>
        </BFormGroup>

        <BFormGroup
          id="length-range-group"
          label="Length range (seconds):"
          label-for="length-range"
        >
          <BFormGroup id="length-range">
            <div class="d-flex">
              <BFormInput
                id="length-min"
                v-model.number="lengthMin"
                type="number"
                min="0"
                max="600"
                style="width: 6rem"
              />
              <span class="mx-2 pt-2">to</span>
              <BFormInput
                id="length-max"
                v-model.number="lengthMax"
                type="number"
                min="0"
                max="600"
                style="width: 6rem"
              />
              <div class="invalid-feedback">Length must be between 0 and 600 seconds</div>
            </div>
          </BFormGroup>
        </BFormGroup>

        <BFormGroup id="activity-group" label="By User:" label-for="activity">
          <div class="d-flex">
            <BFormInput
              id="user"
              v-model="displayUser"
              placeholder="UserName or me"
              style="width: 10rem"
              class="me-2"
            />
            <BFormSelect id="activity" v-model="computedActivity" :options="activities" />
          </div>
        </BFormGroup>

        <BFormGroup
          id="services-group"
          class="mx-2 mb-2"
          label="Available on:"
          label-for="services"
        >
          <BFormCheckboxGroup id="services" v-model="services">
            <BFormCheckbox value="A">Amazon</BFormCheckbox>
            <BFormCheckbox value="I">ITunes</BFormCheckbox>
            <BFormCheckbox value="S">Spotify</BFormCheckbox>
          </BFormCheckboxGroup>
        </BFormGroup>

        <BFormGroup id="bonuses-group" class="mx-2 mb-2" label="Bonus content:" label-for="bonuses">
          <BFormCheckboxGroup id="bonuses" v-model="bonuses">
            <BFormCheckbox value="P">Not found in any publisher catalog</BFormCheckbox>
            <BFormCheckbox value="D">Not categorized by dance</BFormCheckbox>
          </BFormCheckboxGroup>
          <template #description>
            <a href="https://music4dance.blog/music4dance-help/subscriptions/">Premium content</a>
          </template>
        </BFormGroup>

        <BFormGroup id="sort-group" label="Sort By:" label-for="sort">
          <BFormSelect id="sort" v-model="sortId" :options="sortOptions" required />
          <BFormRadioGroup id="sort-order" v-model="sortDirection" name="sort-order" class="my-2">
            <BFormRadio value="asc"
              >Ascending (A-Z, Slow-Fast, Newest-Oldest, Shortest-Longest)</BFormRadio
            >
            <BFormRadio value="desc"
              >Descending (Z-A, Fast-Slow, Oldest-Newest, Longest-Shortest)</BFormRadio
            >
          </BFormRadioGroup>
        </BFormGroup>

        <BAlert :model-value="true" variant="success">
          {{ songFilter.description }}
        </BAlert>

        <div class="d-flex justify-content-between w-100 mx-1 mb-2">
          <BButton type="reset" variant="secondary">Reset</BButton>
          <BButton type="submit" variant="primary">Submit</BButton>
        </div>
      </BForm>
      <BCard v-if="showDiagnostics" class="mt-3" header="Form Data Result">
        <pre class="m-0">
searchString = {{ keyWords }}
dances = {{ dances }}
danceQueryItems = {{ danceQueryItems }}
danceConnector = {{ danceConnector }}
excludeDanceTags = {{ excludeDanceTags }}
tempoMin = {{ tempoMin }}
tempoMax = {{ tempoMax }}
lengthMin = {{ lengthMin }}
lengthMax = {{ lengthMax }}
activity = {{ computedActivity }}
services = {{ services }}
sort = {{ sortId }}
order = {{ sortDirection }}
bonus = {{ bonuses }}
tags = {{ tagString }}
user = {{ user }}
displayUser = {{ displayUser }}

filter = {{ songFilter }}
query = {{ songFilter.query }}
        </pre>
      </BCard>
    </div>
  </PageFrame>
</template>
