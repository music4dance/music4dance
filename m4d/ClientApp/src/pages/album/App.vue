<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
