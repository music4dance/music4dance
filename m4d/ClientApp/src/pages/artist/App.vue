<script setup lang="ts">
import { ArtistModel } from "@/models/ArtistModel";
import { SongFilter } from "@/models/SongFilter";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { TypedJSON } from "typedjson";
import { Song } from "@/models/Song";
import { useSongSelector } from "@/composables/useSongSelector";

declare const model_: string;

const model = TypedJSON.parse(model_, ArtistModel)!;

const danceDB = safeDanceDatabase();
const { songs: selected, select: selectSong } = useSongSelector();

const title = `Artist: ${model.artist}`;
const filter = new SongFilter();
const histories = model.histories ?? [];
const songs = histories.map((h) => Song.fromHistory(h));
const danceIds = [
  ...songs.reduce((acc, h) => {
    const ratings = h.danceRatings;
    if (ratings) {
      ratings.forEach((r) => acc.add(r.danceId));
    }
    return acc;
  }, new Set<string>()),
];
const dances = danceIds.map((id) => danceDB.dances.find((d) => d.id === id)!);
</script>

<template>
  <PageFrame id="app" :title="title">
    <p>
      {{ histories.length }} {{ model.artist }} songs with suggestions for dances to dance to them:
      <span v-for="(dance, idx) in dances" :key="dance.id"
        ><span v-if="idx + 1 === dances.length && dances.length > 1"> and </span
        ><a :href="'/dances/' + dance.seoName">{{ dance.name }}</a
        ><span v-if="idx + 1 < dances.length && dances.length > 2">, </span></span
      >
    </p>
    <SongTable
      :histories="histories"
      :filter="filter"
      :hide-sort="true"
      :hidden-columns="['artist', 'length', 'track']"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
