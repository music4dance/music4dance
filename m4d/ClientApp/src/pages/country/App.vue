<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";

const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Country", active: true }];
</script>

<template>
  <PageFrame id="app" title="Country Western Dance Music" :breadcrumbs="breadcrumbs">
    <p>
      Here at music4dance, we constantly seek ways to serve the broader dance community. We're
      considering adding more Country Western dances to our catalog, but we aren't experts in those
      dances. So please <a href="https://music4dance.blog/feedback/">get in touch with us</a> if
      you're a Country Western dancer and would like to see this site provide support for those
      dances in a way similar to how we support Ballroom dances, including a
      <a href="https://www.music4dance.net/dances/ballroom-competition-categories"
        >competition tempo page</a
      >, pages for each of the dances (as we currently have for
      <a href="https://www.music4dance.net/dances/country-two-step">Country Two Step</a>) and
      include those dances in the
      <a href="https://www.music4dance.net/home/counter">tempo counter</a> and
      <a href="https://www.music4dance.net/home/tempi">tempo list</a> pages.
    </p>
    <p>
      We'd especially <a href="https://music4dance.blog/feedback/">like to hear from you</a> if you
      are an expert in Country Western dances and would be willing to help us build these pages.
      We're looking to start from the tempo guidance the
      <a
        href="https://ucwdc.org/wp-content/uploads/2023/01/UCWDC-ProAm-ProPro-and-Couples-Competition-Music-BPMs-2023-2025.pdf"
        >United Country Western Dance Council</a
      >
      and the
      <a href="https://www1.worldcdf.com/pdf/RULES_BOOK.pdf">World Country Dance Federation</a>
      provide, but would be happy to have your input as well.
    </p>
  </PageFrame>
</template>
