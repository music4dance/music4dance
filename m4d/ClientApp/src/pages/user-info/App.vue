<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { type BreadCrumbItem, infoTrail } from "@/models/BreadCrumbItem";
import { ProfileModel } from "@/models/ProfileModel";
import { TypedJSON } from "typedjson";
import { reactive } from "vue";
import { MenuContext } from "@/models/MenuContext";
import { getMenuContext } from "@/helpers/GetMenuContext";
import UserList from "./UserList.vue";
import UserProfile from "./UserProfile.vue";
declare const model_: string;

const model = reactive(TypedJSON.parse(model_, ProfileModel)!);
const breadcrumbs: BreadCrumbItem[] = [...infoTrail, { text: "User Profile", active: true }];
const menuContext: MenuContext = getMenuContext();
</script>

<template>
  <PageFrame id="app" :breadcrumbs="breadcrumbs">
    <h1>
      Info for <em>{{ model.displayName }}</em
      >'s Profile
    </h1>
    <div>
      <UserProfile :model="model" :menu-context="menuContext"></UserProfile>
      <UserList :model="model"></UserList>
    </div>
  </PageFrame>
</template>
