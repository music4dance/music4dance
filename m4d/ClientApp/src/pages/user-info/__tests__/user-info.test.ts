import { describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestPageSnapshot";
import App from "../App.vue";
import { model } from "./model";

describe("User Info", () => {
  test("Renders User Info Page", () => {
    testPageSnapshot(App, model);
  });
});
