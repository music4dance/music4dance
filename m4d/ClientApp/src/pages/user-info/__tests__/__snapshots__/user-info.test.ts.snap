// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Info > Renders User Info Page 1`] = `
"<div><span context=\\"[object Object]\\">MainMenu</span>
  <nav aria-label=\\"breadcrumb\\">
    <nav aria-label=\\"breadcrumb\\" style=\\"padding: 0.25rem 0.5rem;\\">
      <ol class=\\"breadcrumb\\">
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/\\" target=\\"_self\\">music4dance</a></li>
        <li class=\\"breadcrumb-item\\"><a class=\\"\\" href=\\"/home/info\\" target=\\"_self\\">Info</a></li>
        <li class=\\"breadcrumb-item active\\"><span aria-current=\\"location\\">User Profile</span></li>
      </ol>
    </nav>
  </nav>
  <div id=\\"body-content\\" class=\\"container-fluid body-content\\">
    <div class=\\"mt-2\\"></div>
    <div>
      <h1> Info for <em>chicagoarthurmurray</em>'s Profile </h1>
      <div>
        <div>
          <div class=\\"fs-5\\"> This is a proxy user based on a public spotify profile. View chicagoarthurmurray's spotify profile <a href=\\"https://open.spotify.com/user/chicagoarthurmurray\\" target=\\"_blank\\"><span class=\\"me-1\\">here</span><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\" style=\\"vertical-align: top;\\" class=\\"fs-6\\">
                <g fill=\\"currentColor\\" fill-rule=\\"evenodd\\">
                  <path d=\\"M8.636 3.5a.5.5 0 0 0-.5-.5H1.5A1.5 1.5 0 0 0 0 4.5v10A1.5 1.5 0 0 0 1.5 16h10a1.5 1.5 0 0 0 1.5-1.5V7.864a.5.5 0 0 0-1 0V14.5a.5.5 0 0 1-.5.5h-10a.5.5 0 0 1-.5-.5v-10a.5.5 0 0 1 .5-.5h6.636a.5.5 0 0 0 .5-.5\\"></path>
                  <path d=\\"M16 .5a.5.5 0 0 0-.5-.5h-5a.5.5 0 0 0 0 1h3.793L6.146 9.146a.5.5 0 1 0 .708.708L15 1.707V5.5a.5.5 0 0 0 1 0z\\"></path>
                </g>
              </svg></a>. </div>
        </div>
        <div>
          <h2>chicagoarthurmurray's Song Lists</h2>
          <ul>
            <li data-v-d1889c28=\\"\\" class=\\"list-clean-aligned lh-lg\\"><svg data-v-d1889c28=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <g fill=\\"currentColor\\">
                  <path fill-rule=\\"evenodd\\" d=\\"M8 5.5a.5.5 0 0 1 .5.5v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 .5-.5\\"></path>
                  <path d=\\"m10.273 2.513l-.921-.944l.715-.698l.622.637l.89-.011a2.89 2.89 0 0 1 2.924 2.924l-.01.89l.636.622a2.89 2.89 0 0 1 0 4.134l-.637.622l.011.89a2.89 2.89 0 0 1-2.924 2.924l-.89-.01l-.622.636a2.89 2.89 0 0 1-4.134 0l-.622-.637l-.89.011a2.89 2.89 0 0 1-2.924-2.924l.01-.89l-.636-.622a2.89 2.89 0 0 1 0-4.134l.637-.622l-.011-.89a2.89 2.89 0 0 1 2.924-2.924l.89.01l.622-.636a2.89 2.89 0 0 1 4.134 0l-.715.698a1.89 1.89 0 0 0-2.704 0l-.92.944l-1.32-.016a1.89 1.89 0 0 0-1.911 1.912l.016 1.318l-.944.921a1.89 1.89 0 0 0 0 2.704l.944.92l-.016 1.32a1.89 1.89 0 0 0 1.912 1.911l1.318-.016l.921.944a1.89 1.89 0 0 0 2.704 0l.92-.944l1.32.016a1.89 1.89 0 0 0 1.911-1.912l-.016-1.318l.944-.921a1.89 1.89 0 0 0 0-2.704l-.944-.92l.016-1.32a1.89 1.89 0 0 0-1.912-1.911z\\"></path>
                </g>
              </svg><span data-v-d1889c28=\\"\\" class=\\"px-1\\"></span><svg data-v-d1889c28=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <path fill=\\"currentColor\\" d=\\"M12.146.146a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1 0 .708l-10 10a.5.5 0 0 1-.168.11l-5 2a.5.5 0 0 1-.65-.65l2-5a.5.5 0 0 1 .11-.168zM11.207 2.5L13.5 4.793L14.793 3.5L12.5 1.207zm1.586 3L10.5 3.207L4 9.707V10h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.293zm-9.761 5.175l-.106.106l-1.528 3.821l3.821-1.528l.106-.106A.5.5 0 0 1 5 12.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.468-.325\\"></path>
              </svg><span data-v-d1889c28=\\"\\" class=\\"px-1\\"></span><a data-v-d1889c28=\\"\\" href=\\"/song/filterSearch?filter=v2-index--Modified---%2Bchicagoarthurmurray%7Ca\\" class=\\"ms-1\\">Songs that chicagoarthurmurray has edited.</a><span data-v-d1889c28=\\"\\"> (104)</span></li>
            <li data-v-d1889c28=\\"\\" class=\\"list-clean-aligned lh-lg\\"><svg data-v-d1889c28=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <g fill=\\"currentColor\\">
                  <path fill-rule=\\"evenodd\\" d=\\"M5.5 8a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1H6a.5.5 0 0 1-.5-.5\\"></path>
                  <path d=\\"m10.273 2.513l-.921-.944l.715-.698l.622.637l.89-.011a2.89 2.89 0 0 1 2.924 2.924l-.01.89l.636.622a2.89 2.89 0 0 1 0 4.134l-.637.622l.011.89a2.89 2.89 0 0 1-2.924 2.924l-.89-.01l-.622.636a2.89 2.89 0 0 1-4.134 0l-.622-.637l-.89.011a2.89 2.89 0 0 1-2.924-2.924l.01-.89l-.636-.622a2.89 2.89 0 0 1 0-4.134l.637-.622l-.011-.89a2.89 2.89 0 0 1 2.924-2.924l.89.01l.622-.636a2.89 2.89 0 0 1 4.134 0l-.715.698a1.89 1.89 0 0 0-2.704 0l-.92.944l-1.32-.016a1.89 1.89 0 0 0-1.911 1.912l.016 1.318l-.944.921a1.89 1.89 0 0 0 0 2.704l.944.92l-.016 1.32a1.89 1.89 0 0 0 1.912 1.911l1.318-.016l.921.944a1.89 1.89 0 0 0 2.704 0l.92-.944l1.32.016a1.89 1.89 0 0 0 1.911-1.912l-.016-1.318l.944-.921a1.89 1.89 0 0 0 0-2.704l-.944-.92l.016-1.32a1.89 1.89 0 0 0-1.912-1.911z\\"></path>
                </g>
              </svg><span data-v-d1889c28=\\"\\" class=\\"px-1\\"></span><svg data-v-d1889c28=\\"\\" viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                <path fill=\\"currentColor\\" d=\\"M12.146.146a.5.5 0 0 1 .708 0l3 3a.5.5 0 0 1 0 .708l-10 10a.5.5 0 0 1-.168.11l-5 2a.5.5 0 0 1-.65-.65l2-5a.5.5 0 0 1 .11-.168zM11.207 2.5L13.5 4.793L14.793 3.5L12.5 1.207zm1.586 3L10.5 3.207L4 9.707V10h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.293zm-9.761 5.175l-.106.106l-1.528 3.821l3.821-1.528l.106-.106A.5.5 0 0 1 5 12.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.468-.325\\"></path>
              </svg><span data-v-d1889c28=\\"\\" class=\\"px-1\\"></span><a data-v-d1889c28=\\"\\" href=\\"/song/filterSearch?filter=v2-index--Modified---%5C-chicagoarthurmurray%7Ca\\" class=\\"ms-1\\">Songs that chicagoarthurmurray has <em>not</em> edited.</a>
              <!--v-if-->
            </li>
          </ul>
          <div>
            <p> The User Song List feature is still under development. We are considering including lists of dance styles that the user has tagged or possibly including lists of arbitrary tags. Please <a href=\\"https://music4dance.blog/feedback/\\">let us know</a> if you like this feature or have ideas to improve it (or both). </p>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <div id=\\"footer-content\\">
    <hr>
    <footer>
      <p> Â© 2023 - <a href=\\"https://www.music4dance.net\\">Music4Dance.net</a> - <a href=\\"https://www.music4dance.net/home/sitemap\\">Site Map</a> - <a href=\\"https://www.music4dance.net/home/termsofservice\\">Terms of Service</a> - <a href=\\"https://www.music4dance.net/home/privacypolicy\\">Privacy Policy</a> - <a href=\\"https://www.music4dance.net/home/credits\\">Credits</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
